qiime2/metadata/metadata.py

```diff
@@ -99,32 +217,44 @@ def __init__(self, index):
         self._validate_index(index, axis='id')
         self._ids = tuple(index)
 
-        self._artifacts = []
+        self._sources = []
```

Comment:
I think now that we are tracking the sources in their own partitions, we can walk back on this "pooled" approach, and instead just track the lists in private attributes (`self._artifacts`, `self._tsvs`, etc).

Reply:
@thermokarst, for some reason none of your last three comments were showing up for me until just now

Reply:
Should we still have single `@property sources` that returns a tuple of all of those different sources or something similar as well as `@property` methods that return those different sources individually?

Reply:
Additionally, are you talking about adding the actual DataFrames/tsvs themselves to these properties? If so how? The actual `DataFrame` objects go in the `DataFrames` property and then the filepath to the tsv or something goe in the `tsvs` property? Currently the artifacts are `Artifact` objects and everything else is a `Metadata` object.

Reply:
> Should we still have single @property sources that returns a tuple of all of those different sources or something similar as well as @property methods that return those different sources individually?

Maybe, I need to stew on that a bit.

> Additionally, are you talking about adding the actual DataFrames/tsvs themselves to these properties? If so how

I was envisioning either the `Metadata` objects themselves, or, perhaps a list of IDs.

I will write up my thoughts in more detail later.